docs/source/developer_documentation.rst

```diff
@@ -26,20 +27,31 @@ A variety of coding conventions are enforced by automated tools in continuous in
 PyRenew Principles
 ------------------
 
--  Variable naming conventions
+Variable naming conventions
+---------------------------
+-  Use the ``data_`` prefix for (potentially) observed data.
+-  Use the ``_rv`` suffix for random variables.
+-  Use the ``observed_`` for the output of sample statements where ``obs`` is a ``data_`` prefixed object.
+-  Thus, code which may reasonably written like ``infections = infections.sample(x, obs=infections)`` should instead be written ``observed_infections = infections_rv.sample(x, obs=data_infections)``.
+
+
+Class conventions
+-----------------
+-  Composing ``Model`` objects is discouraged. To modularize and reuse code across ``Model``s, create custom ``RandomVariable`` classes.
+-  Returning anything from ``Model.sample`` is discouraged. Instead, sample from models using ``Predictive`` or our ``prior_predictive`` or ``posterior_predictive`` functions.
```

Comment:
But `sample` is what `__call__` calls, so it needs to be defined. I think this is not up to date?

Reply:
`Model.sample`, not `RandomVariable.sample`. @damonbayer originally wrote this, so I don't want to speak for him, but as I understand it, the idea was that `Model.sample` (alias `Model.model`) should always be _defined_ and should make various `RandomVariable.sample()` calls internally (thus defining the actual model), but should itself be a void function (i.e. should return `None`). I believe the idea was to discourage the user from treating models as `RandomVariables` (see previous point about composition) and to encourage the user to use `Model.prior_predictive()` rather than `Model.sample()` for forward simulation.

Reply:
Totally right. Missed the Model bit 🙃